README.rst

```diff
@@ -59,104 +61,102 @@ Sample Output
-    'LEIN_JAR'                          : '/usr/local/lib/lein/self-installs/leiningen-2.9.4-standalone.jar'
-    'M2_HOME'                           : '/usr/share/apache-maven-3.6.3'
-    'PATH'                              : '/opt/hostedtoolcache/Python/3.9.0/x64/bin:/opt/hostedtoolcache/Python/3.9.0/x64:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/opt/pipx_bin:/usr/share/rust/.cargo/bin:/home/runner/.config/composer/vendor/bin:/home/runner/.dotnet/tools:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin'
-    'PERFLOG_LOCATION_SETTING'          : 'RUNNER_PERFLOG'
-    'PIPX_BIN_DIR'                      : '"/opt/pipx_bin"'
-    'PIPX_HOME'                         : '"/opt/pipx"'
-    'POWERSHELL_DISTRIBUTION_CHANNEL'   : 'GitHub-Actions-ubuntu20'
-    'PWD'                               : '/home/runner/work/python-info-action/python-info-action'
-    'RUNNER_OS'                         : 'Linux'
-    'RUNNER_PERFLOG'                    : '/home/runner/perflog'
-    'RUNNER_TEMP'                       : '/home/runner/work/_temp'
-    'RUNNER_TOOL_CACHE'                 : '/opt/hostedtoolcache'
-    'RUNNER_TRACKING_ID'                : 'github_02a261ae-7169-4a49-a99b-3dc10b5ffe62'
-    'RUNNER_USER'                       : 'runner'
-    'RUNNER_WORKSPACE'                  : '/home/runner/work/python-info-action'
-    'SELENIUM_JAR_PATH'                 : '/usr/share/java/selenium-server-standalone.jar'
-    'SHLVL'                             : '1'
-    'SWIFT_PATH'                        : '/usr/share/swift/usr/bin'
-    'USER'                              : 'runner'
-    'VCPKG_INSTALLATION_ROOT'           : '/usr/local/share/vcpkg'
-    '_'                                 : '/opt/hostedtoolcache/Python/3.9.0/x64/bin/python'
-    'pythonLocation'                    : '/opt/hostedtoolcache/Python/3.9.0/x64'
+    'ACTION_FILE_PATH'                    : 'output_pre.log'
+    'AGENT_TOOLSDIRECTORY'                : '/opt/hostedtoolcache'
+    'ANDROID_HOME'                        : '/usr/local/lib/android/sdk'
+    'ANDROID_NDK_HOME'                    : '/usr/local/lib/android/sdk/ndk-bundle'
+    'ANDROID_NDK_LATEST_HOME'             : '/usr/local/lib/android/sdk/ndk/22.0.7026061'
+    <snip>
+    'SWIFT_PATH'                          : '/usr/share/swift/usr/bin'
+    'USER'                                : 'runner'
+    'VCPKG_INSTALLATION_ROOT'             : '/usr/local/share/vcpkg'
+    '_'                                   : '/opt/hostedtoolcache/Python/3.9.2/x64/bin/python'
+    'pythonLocation'                      : '/opt/hostedtoolcache/Python/3.9.2/x64'
 
     Installed Packages
     ==================
 
-    pip==20.2.4
+    pip==21.0.1
     setuptools==49.2.1
 
+    Workflow Details
+    ================
+
+
+    Steps
+    -----
+
+    {}
+
+    GitHub
+    ------
+
+    {
+        "token": "v1.0c46741a32436f591105a1c12ee7000683bfb58e",
```

Comment:
Was this token present in the output of the action?

I double checked and the token is hidden in the output.

```
      "token": "***", 
```
If this is the real github API token used by the action, and it is displayed in plain text in the output, then this might be a security bug.

------------

```suggestion
        "token": "***",
```

Reply:
I just checked the latest build and as you say it has been masked. Both in the normal log view and the raw log view.

![image](https://user-images.githubusercontent.com/543719/111873971-5674d200-8969-11eb-8406-daae2da3023e.png)
![image](https://user-images.githubusercontent.com/543719/111873975-596fc280-8969-11eb-99a0-e5266705129d.png)

As could reasonably be expected (though sure, I forgot about this) it is \_not\_ masked in the log file the action generates.  Apparently that's where I grabbed this from.  This should definitely be noted in the readme, at a minimum.  Perhaps it should be actively masked by our own code.

https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context
> A token to authenticate on behalf of the GitHub App installed on your repository. This is functionally equivalent to the GITHUB_TOKEN secret. For more information, see "Authenticating with the GITHUB_TOKEN."

https://docs.github.com/en/actions/reference/authentication-in-a-workflow#about-the-github_token-secret
> Before each job begins, GitHub fetches an installation access token for the job. The token expires when the job is finished.

I did check and the latest CPython 3.9 Linux log does have a different token as I would expect.  So, as long as `github.token` expires "functionally equivalent" to `GITHUB_TOKEN` then there isn't any \_lasting\_ security hazard by this being shown now.  I would think that artifacts are the most likely escape hole for the logs after the regular GitHub logs.  Since they are not available until after all the builds complete it would seem that this is not a flagrant security issue that we should be up in arms about.  But, I would like more consideration of how seriously it should be taken and what steps ought to be taken to mitigate it.  I'll raise it in #twisted-dev.

Thanks for pointing this out @adiroiban.

Reply:
True.  Thanks for the analysis.

You can always write the token in a file and upload it the Internet.
You don't need github-info-action for that :) 

So at this point, I think that is enough just to add a warning.